### PR TITLE
Unit test and minor fix 

### PR DESCRIPTION
**Pull request by fiona-gladwin**
- Add support to test FileSourceReader with Brightness and CMN
- Add Brightness fixed support
- Grayscale fix for supported datalaoder

MIVisionX-data branch : [link](https://github.com/fiona-gladwin/MIVisionX-data/tree/TensorGoldenOutputs/rocal_data/GoldenOutputsTensor)

### CODE REVIEW

File: rocAL/rocAL/include/api/rocal_api_augmentation.h
Comment (rrawther) [inline rocAL/rocAL/include/api/rocal_api_augmentation.h:152]: please change the comment to tensor

File: rocAL/rocAL/include/api/rocal_api_augmentation.h
Comment (rrawther) [inline rocAL/rocAL/include/api/rocal_api_augmentation.h:158]: I thnik we need to modify it to support different alpha and beta for each image in tensor, in other words alpha and beta has to be an array instead

File: rocAL/rocAL/source/api/rocal_api_augmentation.cpp
Comment (LakshmiKumar23) [inline rocAL/rocAL/source/api/rocal_api_augmentation.cpp:677]: @fiona-gladwin Are you commenting all the augmentations not implemented? Where is this comment section ending?

**Review by LakshmiKumar23 — CHANGES_REQUESTED** [review]

File: rocAL/rocAL/source/api/rocal_api_augmentation.cpp
Comment (rrawther) [inline rocAL/rocAL/source/api/rocal_api_augmentation.cpp:649]: same comment as before for alpha and beta

File: rocAL/rocAL/source/api/rocal_api_data_loaders.cpp
Comment (rrawther) [inline rocAL/rocAL/source/api/rocal_api_data_loaders.cpp:189]: we need to pass layout instead of color_format in the API for tensor.

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: why is this changed to 0?

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: are we using ROCAL_USER_GIVEN_SIZE in any apis? If not we should rename ROCAL_USE_USER_GIVEN_SIZE_RESTRICTED as ROCAL_USE_USER_GIVEN_SIZE and get rid of RESTR..

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: color_format is not suirable with tensor. Please use layout instead

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: please don't name tensor as image. Change all such naming conventions

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: I would suggest different values for mean and std to test the function with more meaningful parameters

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp:834]: why is this commented?

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp:871]: do we really want to print out all bbox values here?

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: please extract all dims in one call and use it instead of calling at for each dims

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp:802]: we can't mix and match Image and Tensor stuff

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (rrawther) [inline utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: Are we writing each image in a tensor as a separate file? We need to write/display the entire tensor  as a single output. Only one cv::mat  needs to be created per tensor. Please rewrite this code. It looks very messy

**Review by rrawther — CHANGES_REQUESTED** [review]
please address review comments

File: rocAL/rocAL/include/api/rocal_api_augmentation.h
Comment (fiona-gladwin) [reply rocAL/rocAL/include/api/rocal_api_augmentation.h:152]: Brightness augmentation is applied only for images hence we have added the comment stating it controls contrast of the image
Since alpha is used to modify the brightness of images we have added the comment wrt image.

File: rocAL/rocAL/source/api/rocal_api_data_loaders.cpp
Comment (fiona-gladwin) [reply rocAL/rocAL/source/api/rocal_api_data_loaders.cpp:189]: The color format is required to distinguish the format of the image whether it is of RGB, BGR, Grayscale or planar. This argument needs to be passed to the decoder as well.

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: The color format needs to be passed to the decoder which specifies whether it is RGB, BGR or Grayscale formats.

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp:834]: OneHot labels support is not added yet in tensor branch. Hence commenting out temproarily

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: We are extracting the value N from dims() API only once
We are extracting the h and w values from the shape() API here

We planned to use the shape() API to get the width and height since the position of the width and height values in the dims vector will change based on the tensor layout. Instead of checking the layout and obtaining the w and h every time the shape() API is used here which stores only width and height in case of images

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: We are writing the every batch i.e a single tensor into a single jpeg file.
Since there can be multiple output tensors in the pipeline we have created multiple cv matrix for every outputs.

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp:871]: Since for classification we display the labels. Planned to add changes in the detection pipeline to print labels and bbox as well, so we can verify outputs please let us know if this can be retained

File: rocAL/rocAL/source/api/rocal_api_augmentation.cpp
Comment (fiona-gladwin) [reply rocAL/rocAL/source/api/rocal_api_augmentation.cpp:677]: Yes currently we have commented augmentations without tensor support we will enable them in the upcoming PR.
The comment section ending is [here ](https://github.com/LakshmiKumar23/MIVisionX/blob/5447f0afe4e5296f36aeb2e1f07fd365c4558f5f/rocAL/rocAL/source/api/rocal_api_augmentation.cpp#L1672)

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: We will pass different values and regenerate Golden outputs for the same

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: Done

File: rocAL/rocAL/include/api/rocal_api_augmentation.h
Comment (shobana-mcw) [reply rocAL/rocAL/include/api/rocal_api_augmentation.h:158]: If we pass alpha and beta as an array it will be the same for every iteration even though its different for every image in a batch.
If we need to have different value for each image then we need to pass it as rocALTensor and update the values in python using external_source.

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: As discussed. Will issue this change as a separate PR

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: Reverted back

File: utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp
Comment (fiona-gladwin) [reply utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp]: Done

**Comment by fiona-gladwin** [comment]
@rrawther I have addressed all PR comments please check.

File: rocAL/rocAL/include/api/rocal_api_augmentation.h
Comment (rrawther) [reply rocAL/rocAL/include/api/rocal_api_augmentation.h:158]: Ok. For this version we will merge this since it is the same as image based functionality

**Review by rrawther — APPROVED** [review]

**Comment by LakshmiKumar23** [comment]
@rrawther I need to merge PR29 before merging this. Waiting on your approval

**Pull request merged**
